shared/src/common/types/user/users-with-details.type.ts

```diff
@@ -0,0 +1,16 @@
+type UsersWithDetails = {
+  id: number;
+  email: string;
+  createdAt: string;
+  userDetails: {
+    avatarUrl: string | null;
+    createdAt: string;
+    dateOfBirth: string | null;
+    fullName: string;
+    gender: string | null;
+    id: number;
+    userId: number;
+  };
```

Comment:
If it's a full UserDetails obj type, then you could declare it separately in `user-details` folder and use it here